### PR TITLE
Add unit tests for mis-indented expression body functions

### PR DESCRIPTION
**Pull request by 0x6675636b796f75676974687562**
### What's done:

 * Added unit tests for mis-indented expression body functions.
 * See #1330.

### CODE REVIEW

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
+     *   content. The trailing newline, if any, **won't be read** as a separate
+     *   empty string. So, if the content to be read from this file is expected
+     *   to be terminated with an empty string (which is the case if
+     *   `newlineAtEnd` is `true`), then the file should end with **two**
+     *   consecutive linebreaks.
+     * @param testFileStr the name of the resource which has the original content.
+     * @param trimLastEmptyLine whether the last (empty) line should be
+     *   discarded when reading the content of [testFileStr].
+     * @return `true` if transformed file equals expected result, `false` otherwise.
+     * @see compareFilesFromFileSystem
```
Comment (petertrr) [inline diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt:36]: I believe in KDocs preferred way of linking other declarations is via square brackets, then IDEA will also check if this symbol can be resolved

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/util/LintTestBase.kt [hunk 2]
```diff
@@ -22,26 +22,48 @@ open class LintTestBase(private val ruleSupplier: (rulesConfigList: List<RulesCo
...
+     *
+     * @param code the code to check.
+     * @param rulesConfigList an optional override for `this.rulesConfigList`.
+     * @param fileName an optional override for the file name.
+     * @return the list of lint errors.
+     * @see lintMethod
+     */
+    @OptIn(FeatureInAlphaState::class)
+    protected fun lintResult(
+        @Language("kotlin") code: String,
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/util/LintTestBase.kt:48]: What does this annotation do? Will IDEA highlight strings used as this function parameter as if it was Kotlin code?

**Review by petertrr — APPROVED** [review]

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt [hunk 3]
```diff
@@ -37,40 +48,79 @@ class TestComparatorUnit(private val resourceFilePath: String,
...
     /**
-     * @param fileName
-     * @return file content as a list of lines
+     * @param file the file whose content is to be read.
+     * @return file content as a list of lines, or an empty list if an I/O error
+     *   has occurred.
      */
-    private fun readFile(fileName: String): List<String> {
-        var list: List<String> = ArrayList()
+    private fun readFile(file: Path): List<String> =
```
Comment (nulls) [inline diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt:117]: maybe move to companion as in another class?

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/util/LintTestBase.kt (hunk 2 above)
Comment (0x6675636b796f75676974687562) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/util/LintTestBase.kt:48]: Yes, exactly.

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt (hunk 1 above)
Comment (0x6675636b796f75676974687562) [reply diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt:36]: Indeed, except for the `@see` tags, where `[]` are unnecessary.

**Pull request merged**